pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -546,7 +611,7 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 
 				Context("when MAPI machine with the same name and status.authoritativeAPI set to ClusterAPI is created", func() {
 					BeforeEach(func() {
-						mapiMachine = mapiMachineBuilder.WithName(capiMachine.Name).WithAuthoritativeAPI(mapiv1beta1.MachineAuthorityClusterAPI).Build()
+						mapiMachine = mapiMachineBuilder.WithGenerateName("").WithName(capiMachine.Name).WithAuthoritativeAPI(mapiv1beta1.MachineAuthorityClusterAPI).Build()
```

Comment:
Why explicitly cancelling the generate name? Does an object created with both a name and generate name work?